### ISSUE TITLE
Uncaught TypeError: undefined is not a function

### ISSUE THREAD

**Issue by tacionerybemind** [atom-community/autocomplete-plus#219]
Uncaught TypeError: undefined is not a function

**Atom Version**: 0.158.0
**System**: linux 3.13.0-44-generic
**Thrown From**: [autocomplete-plus](https://github.com/atom-community/autocomplete-plus) package, v2.0.3

It started happening when I opened an SQL file and started to type something...

### Stack Trace

```
At /home/thnery/.atom/packages/autocomplete-plus/lib/fuzzy-provider.coffee:165

TypeError: undefined is not a function
  at FuzzyProvider.module.exports.FuzzyProvider.settingsForScopeDescriptor (/home/thnery/.atom/packages/autocomplete-plus/lib/fuzzy-provider.coffee:165:22)
  at FuzzyProvider.settingsForScopeDescriptor (/home/thnery/.atom/packages/autocomplete-plus/lib/fuzzy-provider.coffee:1:1)
  at FuzzyProvider.module.exports.FuzzyProvider.getCompletionsForCursorScope (/home/thnery/.atom/packages/autocomplete-plus/lib/fuzzy-provider.coffee:9:3)
  at FuzzyProvider.getCompletionsForCursorScope (/home/thnery/.atom/packages/autocomplete-plus/lib/fuzzy-provider.coffee:1:1)
  at FuzzyProvider.module.exports.FuzzyProvider.findSuggestionsForWord (/home/thnery/.atom/packages/autocomplete-plus/lib/fuzzy-provider.coffee:151:34)
  at FuzzyProvider.findSuggestionsForWord (/home/thnery/.atom/packages/autocomplete-plus/lib/fuzzy-provider.coffee:1:1)
  at FuzzyProvider.module.exports.FuzzyProvider.requestHandler (/home/thnery/.atom/packages/autocomplete-plus/lib/fuzzy-provider.coffee:62:20)
  at FuzzyProvider.requestHandler (/home/thnery/.atom/packages/autocomplete-plus/lib/fuzzy-provider.coffee:1:1)
  at /home/thnery/.atom/packages/autocomplete-plus/lib/autocomplete-manager.coffee:115:39
  at Array.map (native)
  at AutocompleteManager.module.exports.AutocompleteManager.scatterRequest (/home/thnery/.atom/packages/autocomplete-plus/lib/autocomplete-manager.coffee:114:27)
  at AutocompleteManager.scatterRequest (/home/thnery/.atom/packages/autocomplete-plus/lib/autocomplete-manager.coffee:1:1)
  at AutocompleteManager.module.exports.AutocompleteManager.runAutocompletion (/home/thnery/.atom/packages/autocomplete-plus/lib/autocomplete-manager.coffee:109:6)
  at /home/thnery/.atom/packages/autocomplete-plus/lib/autocomplete-manager.coffee:1:1

```

/cc @atom/core

**Comment by hanjos** [comment]
Same here. I updated **autocomplete-plus** following instructions from #212, and I got the same error, which redirected me here.

**Comment by jonathanTournier** [comment]
I got the same error just after the update

**Comment by joefitzgerald** [comment]
What other packages do you have installed?

**Comment by hanjos** [comment]
`apm list` returns:

Built-in Atom packages (79)
├── archive-view@0.40.0
├── atom-dark-syntax@0.23.0
├── atom-dark-ui@0.42.0
├── atom-light-syntax@0.22.0
├── atom-light-ui@0.36.0
├── autocomplete@0.34.0
├── autoflow@0.20.0
├── autosave@0.19.0
├── background-tips@0.18.0
├── base16-tomorrow-dark-theme@0.22.0
├── base16-tomorrow-light-theme@0.5.0
├── bookmarks@0.31.0
├── bracket-matcher@0.64.0
├── command-palette@0.30.0
├── deprecation-cop@0.20.0
├── dev-live-reload@0.36.0
├── encoding-selector@0.12.0
├── exception-reporting@0.21.0
├── find-and-replace@0.152.0
├── fuzzy-finder@0.62.0
├── git-diff@0.45.0
├── go-to-line@0.27.0
├── grammar-selector@0.40.0
├── image-view@0.44.0
├── incompatible-packages@0.16.0
├── keybinding-resolver@0.24.0
├── language-c@0.33.0
├── language-clojure@0.9.0
├── language-coffee-script@0.38.1
├── language-css@0.24.0
├── language-gfm@0.55.0
├── language-git@0.9.0
├── language-go@0.19.1
├── language-html@0.27.0
├── language-hyperlink@0.12.2
├── language-java@0.13.0
├── language-javascript@0.51.0
├── language-json@0.10.0
├── language-less@0.21.0
├── language-make@0.12.0
├── language-mustache@0.11.0
├── language-objective-c@0.12.0
├── language-perl@0.9.0
├── language-php@0.18.0
├── language-property-list@0.7.0
├── language-python@0.26.0
├── language-ruby@0.45.0
├── language-ruby-on-rails@0.18.0
├── language-sass@0.29.0
├── language-shellscript@0.10.1
├── language-source@0.8.0
├── language-sql@0.11.0
├── language-text@0.6.0
├── language-todo@0.15.0
├── language-toml@0.14.1
├── language-xml@0.25.0
├── language-yaml@0.21.0
├── link@0.28.0
├── markdown-preview@0.112.0
├── metrics@0.40.0
├── notifications@0.21.0
├── open-on-github@0.31.0
├── package-generator@0.34.0
├── release-notes@0.43.0
├── settings-view@0.161.0
├── snippets@0.60.0
├── solarized-dark-syntax@0.29.0
├── solarized-light-syntax@0.13.0
├── spell-check@0.45.0
├── status-bar@0.54.0
├── styleguide@0.36.0
├── symbols-view@0.70.0
├── tabs@0.58.0
├── timecop@0.24.0
├── tree-view@0.139.0
├── update-package-dependencies@0.7.0
├── welcome@0.21.0
├── whitespace@0.27.0
└── wrap-guide@0.27.0

/Users/humberto/.atom/packages (10)
├── atom-terminal@0.6.0
├── autocomplete-plus@2.0.3
├── color-picker@1.4.4
├── git-blame@0.3.3
├── git-history@1.2.0
├── go-plus@3.1.2
├── merge-conflicts@1.2.9
├── nrepl@0.0.6
├── sbrubbles-syntax@0.0.0
└── visual-studio-dark-syntax@0.1.0

**Comment by joefitzgerald** [comment]
@hanjos and you've definitely restarted atom, right?

**Comment by hanjos** [comment]
I've restarted my Mac, just to be sure. To no avail :(

**Comment by joefitzgerald** [comment]
@hanjos: What is `sbrubbles-syntax`?

**Comment by lierdakil** [comment]
You might notice that atom.config.getAll() was introduced in 0.166.0, and initial report seems to be about older version.

**Comment by joefitzgerald** [comment]
Ahh, thankyou @lierdakil! Great point, @hanjos, @jonathanTournier please ensure you have updated to the latest version of atom.

**Comment by hanjos** [comment]
Now it works! My current `atom --version` is `0.176.0`; apparently Atom finally decided to update itself :)

**Comment by hanjos** [comment]
Can't you set the minimum acceptable Atom version in the package?

**Comment by joefitzgerald** [comment]
Alas, no - not yet: https://github.com/atom/atom/issues/3741

We'd like to! But honestly I wasn't aware of the incompatibility with < `0.166.0`. Not sure we will provide a fix for older versions other than "upgrade atom, please" :smile:.

**Comment by joefitzgerald** [comment]
I will leave this open for the time being to catch anyone else who runs into the same issue and find themselves clicking the button to file an issue. 

> TL;DR;? Update Atom, please :smile:

**Comment by jonathanTournier** [comment]
I updated atom ( version 0.176.0 ) , and it's ok, thank's
